samples/web/wcf-callbacks/Version_5/WcfCallbacks/CallbackService.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.ServiceModel;
+using System.Threading.Tasks;
+using NServiceBus;
+
+[ServiceBehavior(
+    InstanceContextMode = InstanceContextMode.Single,
+    Name = "CallbackService")]
+class CallbackService<TRequest, TResponse> : ICallbackService<TRequest, TResponse>
+{
+    IBus bus;
+
+    public CallbackService(IBus bus)
+    {
+        this.bus = bus;
+    }
+
+    public async Task<TResponse> SendRequest(TRequest request)
+    {
+        if (typeof(TResponse).IsEnum)
+        {
+            ICallback enumLocal = bus.SendLocal(request);
+            return await enumLocal.Register<TResponse>();
+        }
+        if (typeof(TResponse) == typeof(int))
+        {
+            ICallback intLocal = bus.SendLocal(request);
+            object intValue = await intLocal.Register();
+            return (TResponse)intValue;
+        }
+        return await bus.SendLocal(request).Register(ar =>
+        {
+            object[] messages = ar.Messages;
+            return (TResponse)messages[0];
+        });
```

Comment:
So because these are the only 3 types of callbacks in the NServiceBus API, these 3 cases are all that would be needed? In other words, the customer would just use this class as is and wouldn't have to add additional boilerplate here. Correct?